File: apps/admin_web/lib/repositories/walletRepository.ts
import { db } from '../firebaseAdmin';

export async function getCreatorWalletSummary(creatorId: string) {
  const balanceSnap = await db
    .collection('walletBalances')
    .where('creatorId', '==', creatorId)
    .limit(1)
    .get();
  const ledgerSnap = await db
    .collection('walletLedger')
    .where('creatorId', '==', creatorId)
    .limit(100)
    .get();
  return {
    balance: balanceSnap.empty ? null : { id: balanceSnap.docs[0].id, ...balanceSnap.docs[0].data() },
    transactions: ledgerSnap.docs.map((doc) => ({ id: doc.id, ...doc.data() })),
  };
}

File: apps/mobile_flutter/lib/features/wallet/repositories/wallet_repository.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import '../../../core/firebase.dart' as fb;
import '../models/wallet_balance_model.dart';
import '../models/wallet_transaction_model.dart';

class WalletRepository {
  final FirebaseFirestore? _firestore;

  WalletRepository({FirebaseFirestore? firestore})
      : _firestore = firestore ?? fb.tryFirebaseFirestore();

  Stream<WalletBalanceModel?> watchBalance(String userId) {
    final fs = _firestore;
    if (fs == null) return const Stream.empty();
    return fs
        .collection('walletBalances')
        .doc(userId)
        .snapshots()
        .map((doc) {
      if (!doc.exists) return null;
      return WalletBalanceModel.fromMap(doc.data()!);
    });
  }

  Stream<List<WalletTransactionModel>> watchTransactions(String userId) {
    final fs = _firestore;
    if (fs == null) return const Stream.empty();
    return fs
        .collection('walletTransactions')
        .where('userId', isEqualTo: userId)
        .orderBy('createdAt', descending: true)
        .limit(50)
        .snapshots()
        .map((snap) => snap.docs
            .map((doc) => WalletTransactionModel.fromMap(doc.data()))
            .toList());
  }
}